adal/build.gradle

```diff
@@ -0,0 +1,151 @@
+apply plugin: 'com.android.library'
+// This plugin publishes adal in to the local maven repo
+apply plugin: 'com.github.dcendents.android-maven'
+//These plugins are used for static code analysis tools
+apply plugin: 'findbugs'
+apply plugin: 'pmd'
+apply plugin: 'checkstyle'
+
+group = 'com.microsoft.aad'
+version = '1.1.16'
+
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+            classpath 'com.github.dcendents:android-maven-gradle-plugin:1.3'
+    }
+}
+
+android {
+    compileSdkVersion 21
+    buildToolsVersion "21.1.2"
+
+    defaultConfig {
+        minSdkVersion 18
+        targetSdkVersion 21
+        versionCode 1
+        versionName "1.1.16"
+        project.archivesBaseName = "adal"
+        project.version = android.defaultConfig.versionName
+    }
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+    sourceSets {
+        main {
+            manifest.srcFile 'src/main/AndroidManifest.xml'
+        }
+    }
+
+    lintOptions {
+        abortOnError false
+    }
+
+    libraryVariants.all { variant ->
+        variant.outputs.each { output ->
+            def outputFile = output.outputFile
+            if (outputFile != null && outputFile.name.endsWith('.aar')) {
+                def fileName = "${archivesBaseName}-${version}.aar"
+                output.outputFile = new File(outputFile.parent, fileName)
+            }
+        }
+    }
+
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```

Comment:
Well, I simply don't know why we need to lib folder. Do we have any particular jar to compile that cannot be grabbed from those central repos? The reason we have today AFAIK is for eclipse setup(In eclipse we set up a android project but not maven project, you have to put those jars somewhere, so eclipse will know their existence). But now we move to android studio, gradle build is integrated. Correct me if i'm wrong.